client/network_status.go

```diff
@@ -0,0 +1,152 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/pkg/server"
+	"github.com/skupperproject/skupper/pkg/utils"
+	"time"
+)
+
+func (cli *VanClient) NetworkStatus() ([]*types.SiteInfo, error) {
+
+	var sites *[]types.SiteInfo
+	var err error
+	err = utils.Retry(5*time.Second, 5, func() (bool, error) {
+		sites, err = server.GetSiteInfo(cli.Namespace, cli.KubeClient, cli.RestConfig)
+		if err != nil {
+			return false, err
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	versionCheckedSites := cli.checkSiteVersion(sites)
+
+	var services *[]types.ServiceInfo
+	err = utils.Retry(5*time.Second, 5, func() (bool, error) {
+		services, err = server.GetServiceInfo(cli.Namespace, cli.KubeClient, cli.RestConfig)
+		if err != nil {
+			return false, err
+		}
+		return true, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var listSites []*types.SiteInfo
+
+	for _, site := range versionCheckedSites {
+
+		if len(site.Namespace) == 0 {
+			return nil, errors.New("unable to provide site information")
```

Comment:
Suggest use of fmt.Errorf in place of errors.New, has generally been the convention used.

Reply:
Sure thing, I agree